contracts/src/v0.8/automation/dev/v2_3/AutomationRegistryBase2_3.sol

```diff
@@ -1008,6 +1020,10 @@ abstract contract AutomationRegistryBase2_3 is ConfirmedOwner {
       IERC20 token = billingTokens[i];
       BillingConfig memory config = billingConfigs[i];
 
+      // if LINK is a billing option, payout mode must be ON_CHAIN
+      if (address(token) == address(i_link) && s_payoutMode == PayoutMode.OFF_CHAIN) {
+        revert InvalidBillingToken();
+      }
```

Comment:
IIRC, we started reading the state vars in the loop partly to avoid stack to deep errors, right? Could we add a `@dev` comment explaining that so that people don't think we're lazy lolol.